packages/MapFactory/src/DTK_C_API_Map.hpp

```diff
@@ -83,10 +85,45 @@ struct DTK_MapImpl : public DTK_Map
             target_nodes.extent( 1 ) );
         Kokkos::deep_copy( target_nodes_copy, target_nodes );
 
-        // Create a nearest neighbor operator.
-        _map = std::unique_ptr<NearestNeighborOperator<map_device_type>>(
-            new NearestNeighborOperator<map_device_type>(
-                teuchos_comm, source_nodes_copy, target_nodes_copy ) );
+        auto const which_map =
+            ptree.get<std::string>( "Map Type", "Undefined" );
+        if ( which_map == "Undefined" )
+            throw DataTransferKitException(
+                R"(Field "Map Type" is not defined in options string argument for map creation)" );
+        else if ( which_map == "Nearest Neighbor" || which_map == "NN" )
+            _map = std::unique_ptr<NearestNeighborOperator<map_device_type>>(
+                new NearestNeighborOperator<map_device_type>(
+                    teuchos_comm, source_nodes_copy, target_nodes_copy ) );
+        else if ( which_map == "Moving Least Squares" || which_map == "MLS" )
+        {
+            // NOTE if field "Order" is misspelled (for instance first letter
+            // not capitalized), the default value (linear polynomials) will be
+            // picked up without a warning or an error being raised.
+            auto const order = ptree.get<std::string>( "Order", "Linear" );
+            if ( order == "Linear" || order == "1" )
+                _map = std::unique_ptr<MovingLeastSquaresOperator<
+                    map_device_type, Wendland<0>,
```

Comment:
I'm OK with using a Wendland basis for now. Should we default to something higher order (i.e. `Wendland<2>` or `Wendland<4>`)?

Reply:
Let's default to something that is fast.